scripts/fiximports/main.go

```diff
@@ -39,49 +52,152 @@ func main() {
 			!strings.HasSuffix(info.Name(), ".go"):
 			return nil
 		}
-		return fixGoImports(path)
+		files = append(files, path)
+		return nil
 	}); err != nil {
-		fmt.Printf("Error fixing go imports: %v\n", err)
+		_, _ = fmt.Fprintf(os.Stderr, "Error walking directory: %v\n", err)
 		os.Exit(1)
 	}
+
+	// Read all file contents in parallel
+	fileContents := readFilesParallel(files, numWorkers)
+
+	// Because we have multiple ways of separating imports, we have to imports.Process for each one
+	// but imports.LocalPrefix is a global, so we have to set it for each group and process files
```

Comment:
This was the reason I did not make the script execute in parallel. Per group parallelism makes sense 👍

Reply:
My first naive attempt when I dove in here tried for complete parallelism but I was quickly educated about why you didn't do it in the first place!